x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/40_semantic_text_query.yml

```diff
@@ -878,3 +878,117 @@ setup:
 
   - match: { hits.total.value: 1 }
   - match: { hits.hits.0._id: "doc_1" }
+
+---
+"Query using a sparse embedding model with size set to zero":
```

Comment:
Could we please also have a test for 0 hits because the field exists but no embeddings exist?

Reply:
Or better yet, ensure some docs exist and others don't, and we return the correct number of hits?

Reply:
This currently can't happen. _Any_ input, including an empty string, will result in embeddings being generated and indexed. Of course, those embeddings have questionable value. 

We have an issue for handling empty input in a consistent way in the backlog. Once it's addressed, we could potentially add the test you are requesting.

Reply:
We can have documents indexed that don't have that field populated no?

Reply:
Yes, but that's already part of the test

Reply:
Ah you're right. Unfortunate that 0 can't be tested.